content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -102,11 +102,102 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
+### Solution
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array $p$. The subarray sum $a_j \dots a_{i-1}$ where $j\leq i$ is $p[i]-p[j]$, so we are looking for the maximum possible value of $p[i]-p[j]$ over $0 \leq i \leq j \leq N$. 
+
+For a fixed right bound $i$, the maximum subarray sum is
+
+$$$
+p[i]-\min_{j\leq i}{p[j]}
+$$$
+
+Thus, we can keep a running minimum to store $\min_{j\leq i}{p[j]}$ as we iterate through $i$. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
+
+<LanguageSection>
+
+<CPPSection>
+
+C++ code:
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+		// Max subarray sum is the maximum difference between two prefix sums
+		max_subarray_sum = max(max_subarray_sum, dp[i] - min_prefix_sum);
+		min_prefix_sum = min(min_prefix_sum, dp[i]);
+	}
+	cout << max_subarray_sum << endl;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
+
+<Warning>
+
 This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+</Warning>
+
+<Spoiler title = "Alternative Solution: Kadane's Algorithm">
+
+Kadane's Algorithm finds the largest subarray sum using a greedy method. Check the video below for details:
+
+[Video explanation of the max subarray sum using Kadane's Algorithm](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s).
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl x(n);
+	for(int i=0; i<n; i++){
+		cin >> x[i];
+	}
+	ll current_sum = x[0], max_subarray_sum = x[0];
+	for(int i=1; i<n; i++){
+		current_sum = max(current_sum + x[i], x[i]); 
+		// Continue the subarray sum or start a new subarray sum beginning at the current element
```

Comment:
This line is too long, maybe put a line break before "or"?

Reply:
ok.

Reply:
This hasn't been fixed yet...

Reply:
Oops - forgot.